# RNA-seq_pipeline


These are the selection test scripts for the GSOC-project 'Creating a new pipeline for long-read RNA-seq using StringTie v2'

NOTE: Please give co-authorship if you are using or modifying this code!

Approach 1

Stringtie v2 was used to create gtf files,the output generated by tablemaker by using the gtf and bam input files  was used to
create a ballgown object and an FPKM matrix was generated and stored into a csv file.

The bash script bash/_script_w/_tablemaker.sh is used to 
1) Create the gtf files using stringtie v2.
2) Store the output into first1000_brain.gtf and first1000/_liver.gtf in your home folder.
3) Store the output generated by tablemaker in brain_output and liver/_output folders in your home directory.

The output generated by tablemaker is used by the R script and the FPKM matrix is stored in the csv files.

This approach results in some errors like:
1) BAM error: CIGAR op has zero length (ch169_read16467/_template/_fail/_FAH60712)
2) BAM record error: found spliced alignment without XS attribute

To overcome these errors, approach 2 can be used. 

Approach 2

Stringtie v2 was used to create the gtf files and the merging was also performed using stringtie. The merged gtfs were used by stringtie and 
the output was obtained which was the data for ballgown and then the fkpm matrices were obtained using ballgown.

The bash script bash_script/_wo_tablemaker.sh was used to check if there is any existing output files((the files containing data for ballgown)) and in case there is an
existing output files, then the folder containing the files is removed.

This approach doesn't produce any errors and works smoothly.

